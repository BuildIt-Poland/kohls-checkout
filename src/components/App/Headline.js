import styled from 'styled-components';

const Headline = styled.h1`
  font-size: 20px;
  font-weight: 700;
  font-family: sans-serif;
`;

export default Headline;
