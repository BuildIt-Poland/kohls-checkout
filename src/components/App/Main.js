// Whole page wrapper with accomodation for fixed header

import styled from 'styled-components';

const MainContent = styled.main`
  margin-top: 6rem;
`;

export default MainContent;
