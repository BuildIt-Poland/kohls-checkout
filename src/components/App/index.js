import React from 'react';

import Item from '../Item/Item';

function App() {
  return (
    <section>
      <Item />
    </section>
  );
}

export default App;
