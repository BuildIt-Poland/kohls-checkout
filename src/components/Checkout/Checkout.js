import React from 'react';
import { Route } from 'react-router-dom';
import styled from 'styled-components';

import CheckoutDelivery from '../CheckoutDelivery';
import CheckoutPayment from '../CheckoutPayment';
import CheckoutReview from '../CheckoutReview';
import Navigation from '../Navigation';
import { CHECKOUT_DELIVERY, CHECKOUT_PAYMENT, CHECKOUT_REVIEW, CHECKOUT } from '../../constants/routes';

const Wrapper = styled.section`
  margin-top: 100px;
`;

function Checkout() {
  return (
    <>
      <Route path={CHECKOUT} component={Navigation} />
      <Wrapper>
        <Route path={CHECKOUT_DELIVERY} component={CheckoutDelivery} />
        <Route path={CHECKOUT_PAYMENT} component={CheckoutPayment} />
        <Route path={CHECKOUT_REVIEW} component={CheckoutReview} />
      </Wrapper>
    </>
  );
}

export default Checkout;
