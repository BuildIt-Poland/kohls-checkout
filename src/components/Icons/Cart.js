import React from 'react';

function Cart() {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" aria-labelledby="icon-cart-title" width="41" height="37">
      <title id="icon-cart-title">Cart</title>
      <path
        fill="#222"
        d="M16.734 31.968a4.5 4.5 0 1 1-9 0 4.5 4.5 0 0 1 9 0zm10.103-4.5a4.5 4.5 0 1 0 0 9 4.5 4.5 0 0 0 0-9zM9.422 5.598V.532H.837v3.377H6.03V23.69h27.076L40.36 5.598H9.422z"
      />
    </svg>
  );
}

export default Cart;
