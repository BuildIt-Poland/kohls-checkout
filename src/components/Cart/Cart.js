import React from 'react';
import Item from '../Item';

import { CHECKOUT_DELIVERY_PATH } from '../../constants/routes';
import Headline from '../Headline';
import NextStep from '../NextStep';
import OrderSummary from '../OrderSummary';
import Content from './Content';
import Wraper from './Wrapper';

function Cart() {
  return (
    <Wraper>
      <Content as="section">
        <Headline>Cart</Headline>
        <Item />
      </Content>
      <OrderSummary />
      <NextStep label="Checkout" to={CHECKOUT_DELIVERY_PATH} />
    </Wraper>
  );
}

export default Cart;
