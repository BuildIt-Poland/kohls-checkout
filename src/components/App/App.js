import React from 'react';
import { Route } from 'react-router-dom';

import { ROOT_PATH, CHECKOUT_PATH, CART_PATH } from '../../constants/routes';
import Header from '../Header';
import Main from './Main';
import LandingPage from '../LandingPage';
import Cart from '../Cart';
import Checkout from '../Checkout';

function App() {
  return (
    <>
      <Header />
      <Main>
        <Route exact path={ROOT_PATH} component={LandingPage} />
        <Route exact path={CART_PATH} component={Cart} />
        <Route exact path={CHECKOUT_PATH} component={Checkout} />
      </Main>
    </>
  );
}

export default App;
