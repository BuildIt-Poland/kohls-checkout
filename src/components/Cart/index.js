import { connect } from 'react-redux';

import { refillDemoCart } from '../../actions/cartActions';
import Cart from './Cart';

function mapStateToProps(state) {
  return {
    items: state.cart.items
  };
}

function mapDisaptchToProps(dispatch) {
  return {
    refillDemoCart: () => dispatch(refillDemoCart())
  };
}

export default connect(
  mapStateToProps,
  mapDisaptchToProps
)(Cart);
