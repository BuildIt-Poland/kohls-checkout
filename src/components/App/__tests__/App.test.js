/* global getRoutesMap:readonly */

import React from 'react';
import { shallow } from 'enzyme';
import { Route } from 'react-router-dom';

import App from '../App';
import Cart from '../../Cart';
import Checkout from '../../Checkout';

const shallowRender = () => shallow(<App />);

describe('COMPONENTS - App', () => {
  it('renders self and correct routes', () => {
    const wrapper = shallowRender('/cart');
    const routesMap = getRoutesMap(wrapper);

    expect(wrapper.find(Route).length).toBe(2);
    expect(routesMap['/cart']).toBe(Cart);
    expect(routesMap['/checkout/:checkoutStep']).toBe(Checkout);
  });
});
