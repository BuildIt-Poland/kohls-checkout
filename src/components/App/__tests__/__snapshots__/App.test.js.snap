// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - App should render correctly cart route 1`] = `
.c0 {
  font-size: 2rem;
  font-weight: 700;
  font-family: sans-serif;
}

<section>
  <h1
    className="c0"
  >
    Hello world
  </h1>
  <section>
    <h2>
      Cart placeholder
    </h2>
  </section>
</section>
`;

exports[`COMPONENT - App should render correctly checkout route 1`] = `
.c0 {
  font-size: 2rem;
  font-weight: 700;
  font-family: sans-serif;
}

<section>
  <h1
    className="c0"
  >
    Hello world
  </h1>
</section>
`;

exports[`COMPONENT - App should render correctly main route 1`] = `
.c0 {
  font-size: 2rem;
  font-weight: 700;
  font-family: sans-serif;
}

<section>
  <h1
    className="c0"
  >
    Hello world
  </h1>
</section>
`;
